### ISSUE TITLE
Video-related feedback (intro, Sonic Team logo, FMV cutscenes)

### ISSUE THREAD

**Issue by DisapprovingOwl** [PiKeyAr/sadx_dreamcast#175]
@PiKeyAr - So, as promised, here's my first crack at recreating the DC intro from the highest-quality elements we have available. First up, here's a ZIP containing the combined movie (Sonic Team logo + intro) with fades between the two sections as a PNG sequence:

**Combined version with fades** - https://we.tl/Zz0Ct3uqko

Secondly, these ZIPs contain the Sonic Team logo/intro as separate PNG sequences with no baked-in fades - thought I'd better supply these in case you'd prefer to combine them and add fades yourself, or in case you're able to make the game use separate movies with no fades for these in future.

**Sonic Team logo (no fades)** - https://we.tl/WZBwHHcFRK
**Intro (no fades)** - https://we.tl/ddiKJd3d8s

**(Note: Unfortunately the links above are temporary WeTransfer links as I have nowhere to upload these properly, so please grab them in the next 7 days before the links expire...!)**

-----

Thought a log of what's been done/changed might also be useful, so here goes!

Please note, all shots not on the following list are sourced from the GC intro, which seems to win over PC cutscenes quality wise aside from the cropping - many of these have had to be slightly retimed to match the DC intro exactly, but the changes were so minor it wasn't worth adding them to the list. Timecodes in the list below reference the combined version with baked-in fades.

The few DC-sourced bits have been upscaled horizontally in Photoshop using the Preserve Detail 2.0 algorithm, which should hopefully help match them in better with any GC/PC sourced shots.

- **00:06:22** - **Station Square skyline** - mix of DC (first segment with just clouds - useful if we need a version with no fades in future) GC (starting with pan down to buildings) and PC Super Sonic cutscene (to restore the crossfade into the next shot) 
- **00:16:16** - **Station Square shot 1** - PC Super Sonic cutscene
- **00:18:22** - **Station Square shot 2** - PC Super Sonic cutscene
- **00:21:08** - **Station Square shot 3** - PC Super Sonic cutscene during crossfade from previous shot, mixing to GC intro afterwards for best possible quality.
- **00:24:00** - **Station Square shot 4** - GC intro, though I've added an extra white flash effect at 00:26:16 to match the DC version.
- **00:32:27** - **"Sonic Adventure" logo** - DC intro (obviously! = P)
- **00:47:09** - **Building/tidal wave shot 2** - Starts off as GC intro and crossfades throughout shot to PC cutscene. This now exactly matches the DC shot.
- **01:07:02** - **Angel Island shot 2** - DC intro, with highlight detail overlaid from PC cutscene (retimed to match DC intro) on the short section where no extra blurring/effects are visible, in order to restore additional horizontal resolution where possible. The rest has enough blurring/extra effects applied that the quality difference isn't noticeable.
- **01:10:03** - Egg Carrier/Station Square shot - DC intro for the transition effect, PC cutscene thereafter (retimed to match DC intro), intercut with GC intro for the quick cuts back to the forest scene. The transition section is slightly lower quality than GC as a result, but it's thankfully very short so not noticeable in motion - once that's over, the rest of the clip is now HQ and exactly matches the section used in the DC intro.
- **01:12:26** - Two-frame cut to black added here to match the DC intro. Unsure if this was an error in the original DC version - I can replace them with white frames if preferred.
- **01:16:25** - **Tails shot** - A complex one, this: Tails is from the DC intro at all times, matted onto the higher-quality background from the GC intro (modified slightly to mimic a missing fog effect at start of the shot). Once the water bursts out from the windows, this gradually transitions to the full DC shot just before the flash/zoom effect, as after this point the camera angle and effects differ significantly from the GC version. Hopefully the fast movement of Tails and the rushing water should help hide any quality difference, especially when combined with the HQ GC background at the start of the shot.
- **01:29:00** - **Gamma shot** - Another complex one, and probably the shot I'm most pleased with overall. The GC version has problems and lacks vertical resolution, almost like it's been accidentally run through a deinterlacing process. This meant that as well as retiming the shot to match the faster DC version, I also had to interlace fields from both the upscaled DC and GC versions, to preserve the better vertical resolution of the DC shot while gaining added horizontal resolution from GC. Thankfully this seems to have worked and, after some trial and error, I was able to create a proper HQ version of this shot from both sources. Short of having the original source, this is probably the best looking version of this shot we can get!
- **01:36:20** - **Water droplet quick-cut 1** - Replaced the first frame of this shot with a repeat of the second to match the DC intro. I'm unsure if this was an error or a creative change by the original devs due to the combination of fast motion and quick cuts in this sequence - I can revert it to match the GC intro if preferred.
- **01:37:12** - **Water droplet quick-cut 2** - Replaced the first frame of this shot with a repeat of the second to match the DC intro. As above, I'm unsure if this was an error or a creative change by the original devs due to the combination of fast motion and quick cuts in this sequence - I can revert it to match the GC intro if preferred.
- **01:46:09** - **Gamma quick-cut** - GC intro, but with the better-looking "102" from the DC version overlaid onto it.
- **01:52:27** - **Final shot of Perfect Chaos** - Everything after the negative transition effect has been recreated using PC Super Sonic cutscene, in order to mimic the timing/freeze frame seen in the DC intro. It's not a 100% match, as during the GC intro's transition to the negative effect Chaos' movements and the transition itself are both timed slightly differently to DC - the only way to solve this would be to remake the transition too, and I wasn't confident I'd be able to 100% match the DC version if I attempted this. For now, I've got as close to DC as is possible using the DX GC/PC assets, and have made sure the freeze-frame is reached with timing matching the DC version. The hue of the negative effect has also been tweaked to better match DC, which is more orange/purple compared to DX's blue/pink.
- **01:55:09** - Ends on single black frame same as original DC intro - I can change this to an additional frame of the freeze-frame of Perfect Chaos' eye if preferred.

For the full version with fades, I matched the timing for the start of each section to your previous version so that the audio track from your previous version should match up timing-wise. This means there's an additional frame of black at the start, which was hidden by the fade-in on your version, and a few frames gap between the Sonic Team logo and the start of the intro itself. Note that the combined movie is marginally longer than your original version - this is due to matching the intro to the DC version, which has additional frames at the end compared to the GC/PC intro.

I did briefly experiment with seeing if a proper 640x480 version with no black bars was possible, but while it's doable for most shots there's still a few DC-only shots (like the logo, the Tails shot and a couple of other segments) that we have no uncropped source for. Annoyingly enough, some of the Japanese SA TV ads seem to have used uncropped versions of a segment of the Sonic Team logo and the full Tails shot, and possibly some others - if we had access to an HQ source for those adverts then a true 640x480 version of the intro may be more likely to be possible.

Anyway, hope it's all OK - I'm sure some things could be improved even further, so let me know if any tweaks or revisions are needed! = )

**Comment by Windii** [comment]
Pretty cool, but if you're using VirtualDub to combine those stills into a video, make sure you add a chroma smoother filter to smooth out the "pixelly" parts around particularly bright colors. Set the filter to 4:1:0 (Indeo), it's the most effective.

**Comment by DisapprovingOwl** [comment]
Thanks for the tip @Windii - @PiKeyAr will be handling combining the stills back into a video and syncing it all up with the audio. = )

**Comment by DisapprovingOwl** [comment]
@PiKeyAr - Right, as promised on Twitter, here's a restored true 640x480 version of the Sonic Team logo! Thanks for hooking me up with the AutoDemo version to help fill in the missing areas. As before, the upload is a temporary WeTransfer link so please download within the next 7 days!

https://we.tl/yj1ILjdbU2

The restoration here was slightly more complex than anticipated:
- Firstly, the AutoDemo version has a fade in/fade out baked into it - intitially I thought this would mean we'd need to bake in fades to the restored version, but thankfully it seems the original devs were lazy which allowed me to avoid this. I noticed that during the fades the water effect stopped matching up between the two -  turns out rather than re-rendering their clip with no fades, they literally cross-faded clean sections from elsewhere in the video over the fades to hide them. I was able to do the same, and after a little trial and error replicating the fades correctly I managed to create a version of the extra top/bottom areas of the image with no fade in/out baked in.
- Secondly, it turned out that upscaling the AutoDemo version to 640x480 didn't work as expected - the two matched up visually, but the AutoDemo version seemed stretched horizontally compared to the final. Turns out that while the AutoDemo version does show more vertical area, it's actually cropped slightly horizontally and simply stretched to fill the frame. Reducing it to 90% width matched it up exactly with the final version, but this presented its own problem as it left a 32x16px rectangle of blank space in each corner of the video! Happily, the animated water background is simple enough that running the frames through a content-aware fill algorithm to plug the gaps seems to have worked convincingly, resulting in a usable full-frame 640x480 version of the video. Phew!

Since you said on Twitter you were planning on including split versions of the logo/intro in the mod if this worked, the new ZIP I've uploaded only contains the new 640x480 Sonic Team logo sequence with no fades. If you've not already grabbed it, a separate version of the recreated intro movie with no fades is linked near the top of my original post above.

Hope that's all good - as before, if you spot anything with either video which needs tweaking/changing, just let me know and I'll give it a look. = )

**Comment by DisapprovingOwl** [comment]
@PiKeyAr - As discussed on Twitter, here's a revised version of the Tails shot:

https://we.tl/C8hFJ5ep2a

For this one, I've basically started again from scratch, drawing the masks frame-by-frame rather than relying on keyframed masks as before. This has been a bit more time-consuming, but produced much better and more accurate results - much more background detail has been restored than before, and Tails' tails and feet are now properly masked the whole way through which wasn't always the case in the previous version. I've also overlaid shadow detail from the GC shot on some areas of the shot where required, to try and restore as much detail in each frame as possible. As before, I've had to transition to full DC footage just before the white flash, but I've made this a simple crossfade and much shorter than before, both to look less obvious and keep the HQ GC version backgrounds for as long as possible before the camera work etc. start to differ.

Hope it's OK - put a lot more work into this one, but I'm pleased with how it's turned out, so hopefully it's good enough to use! = )

**Comment by PiKeyAr** [comment]
@DisapprovingOwl Like I said on Twitter, great work! This is some impressive restoration that I wouldn't have been able to do on my own.
Here are MPEG1 encodings of the intro as they would appear in the mod:
Intro movie: https://dcmods.unreliable.network/owncloud/data/PiKeyAr/files/Various/intro_f3.mpg
Sonic Team logo: https://dcmods.unreliable.network/owncloud/data/PiKeyAr/files/Various/sonicteam_f.mpg
(the fade transitions will be handled by the mod)
Let me know what you think about the quality of the video. I made sure there was a decent overhead so the intro should be about as good as the Gamecube intro quality wise.

@Windii I generally prefer to avoid using any filters because I assemble the video manually from still frames using ffmpeg. Here's a link to the uncompressed video: https://mega.nz/#!Y0tHkIRR!k48EMZGzyZ45TWbW33DvvddepUeC0Zylo65T_VYd5tI 
Feel free to mess around with it, and if you come up with a higher quality version I'll reencode it for use in the mod.

**Comment by RinMaru** [comment]
The quality is pretty good a bit noisy in some parts but way better then before

**Comment by DisapprovingOwl** [comment]
@PiKeyAr - Quality looks good to me - great work! Looking forward to seeing it in-game. = D

@RinMaru - I think much of the noise in there is due to the original source we've used to reconstruct this - the majority of shots are sourced from the GC intro, which seems to be a little noisier than the other sources used (particularly noticeable on any Station Square shots) but also has the highest amount of detail overall, as well as the best quality encode. I've used this as the primary source and mixed in DC/PC sourced elements wherever they were needed.

**Comment by PiKeyAr** [comment]
@DisapprovingOwl's Sonic Team logo and intro movie recreations are now included in the mod by default, along with a few new options in the config. Please report issues, if any.

**Comment by cheatfreak47** [comment]
I mentioned this on IRC but I noticed that the lower the resolution is of the game, the darker the videos are, this is unlikely to be the intended behavior, so uh, you may need to figure that out I guess.

**Comment by PiKeyAr** [comment]
That's technically impossible because the "darkening" comes from changing the video's default color, which remains the same at all resolution. One potential cause could be the FMV fadeout feature, but it also wouldn't be dependent on resolution.

**Comment by Windii** [comment]
Alright, here's my own encode of the opening with chroma smoothing added. It's also uncompressed.
https://drive.google.com/open?id=1DcKaTkC1TWZkSThC37jyLg0sHLcDunHg

The differences are minimal, since as I've said the filter mainly affects the brightly-colored corners, but it really does help a lot, especially on the area around Tails's eyes in his shot.

**Comment by PiKeyAr** [comment]
@Windii Here's your video encoded with the same settings as @DisapprovingOwl's original intro: https://dcmods.unreliable.network/owncloud/data/PiKeyAr/files/Various/b3.mpg
Here's the original for comparison:
https://dcmods.unreliable.network/owncloud/data/PiKeyAr/files/Various/b4.mpg
I did notice the improvement in Tails' closeup shots, but I also noticed stuff like this:
![image](https://user-images.githubusercontent.com/23312549/37759290-810d9442-2df6-11e8-82de-083c94fb6fe5.png)
vs
![image](https://user-images.githubusercontent.com/23312549/37759295-85ed461a-2df6-11e8-909d-ca68c53b7b29.png)
Knuckles' fur texture looks more washed out in the top screenshot, which is from the video with the filter.
I also noticed that your AVI file is about 800MB, whereas the video I assembled from @DisapprovingOwl's stills is 1.2GB. You sure it doesn't have some kind of compression? If you could get me some still frames with the filter applied, I could add them to the intro without recompression, and compress the final result.

**Comment by Windii** [comment]
It's not compressed. It uses a lossless Ut Video codec. There's no difference in quality at all.

I could just send a video file of just the Tails shot with chroma smoothing on if you want.

**Comment by PiKeyAr** [comment]
Just a list of scenes where you think the filter works well would do. If your video is indeed lossless, I'll just extract the frames from it and insert into the one I use for final encoding.

**Comment by Thesch28** [comment]
The new intro looks amazing!

**Comment by Windii** [comment]
Here's just a Tails clip with chroma smoothing on. The effect works best on him. Other shots appear too fast for the point-upsampled chroma to really be a noticeable issue.
https://drive.google.com/open?id=1hkpzNHCJufOBclZWto_spsXjMOVoSjGx

**Comment by Windii** [comment]
Looks pretty good to me.

**Comment by PiKeyAr** [comment]
The intro in both games uses 192kb audio. I originally wanted to use the SADXPC audio because it's also 48000hz whereas the SA1 intro is 44100hz but there were some timing issues so I just converted the original intro's ADX soundtrack to the same format used by the SADX intro. The conversion wasn't 100% lossless, but I highly doubt there is any perceptible difference between this and the original ADX sound.

**Comment by PiKeyAr** [comment]
The updated intro will be included in the next update.

**Comment by cheatfreak47** [comment]
@PiKeyAr "Technically Impossible" or not, that's what it's doing on my machine.
Which means that's what it's probably doing on some other machines too.

https://gfycat.com/AbleWideeyedAlpinegoat

[Note: I just used the window resize feature to demonstrate it for the gif. 
Running at low resolutions without it yields identical results.]

**Comment by DisapprovingOwl** [comment]
@PiKeyAr - I know this issue's closed, but just thought I'd better post as I have a quick update:

I'm currently completing work on a fixed version of the vanilla SADX intro's longer version of the shot of Gamma firing his gun, to help out the folks working on the HD Cutscenes mod - I'd originally thought this fix impossible, but on closer investigation it turns out most of the differences in timing are due to repeated frames! In fact, there's only one frame not present in both the DC and DX intros - the one right before he fires his gun. Thankfully the AutoDemo intro can just about be used to smooth this frame out too (along with a cheat for the added "102" using fields from an earlier frame at pretty much the same angle).

The reason I'm bringing this up here is that while working on this I've also managed to improve one of the frames in the sequence that's shared with the DC version - one of the frames where Gamma's firing his gun (02488.png in the PNG sequence for the recreated DC intro) has a lot of compression artefacts from the DC version bleeding through. Since there's no detail as such on Gamma's gun, I've painted it out where possible using clean sections from the GC version of the same frame. I know it's only a single frame's difference, but would you be interested in updating the intro with this additional fix? If so, let me know and I'll upload the fixed frame here. = )

Also, a quick question - I remember you mentioned using ffmpeg to convert the PNG stills I supplied back into a usable MPEG. Once I've completed work on the fixed version of the DX Gamma shot, I'm hoping to create a quick mod which just replaces the standard SADX intro with a version with the fixed Gamma shot (using the GC intro with the top/bottom filled in from the PC intro). Your encoding of the intro for DC Conversion struck a great balance quality and filesize-wise - would you be willing to share the settings you used?

**Comment by PiKeyAr** [comment]
Here's the command line I used to convert the uncompressed video into the format SADX2004 works with:
To extract raw frames as PNGs (Sonic Team logo SFD as an example):
```
ffmpeg -i sonicteam.sfd sonicteam\img%04d.png
```
To convert PNGs+WAV audio into a video:
```ffmpeg -framerate 29.97 -i intro\%%05d.png -i intro.wav -s 640x480 -vcodec rawvideo -vf "fps=29.97,format=yuv420p,pad=640:480:(ow-iw)/2:(oh-ih)/2" -acodec pcm_s16le mcal.avi
```
To create the final video from the uncompressed video:
```
ffmpeg -i mcal.avi -pass 1 -vcodec mpeg1video -b:v 6600000 -s 640x480 -acodec mp2 -ab 192k -passlogfile log_file intro_f.mpg
ffmpeg -i mcal.avi -pass 2 -vcodec mpeg1video -b:v 6600000 -s 640x480 -acodec mp2 -ab 192k -passlogfile log_file intro_f.mpg
```
As for the DC intro, I don't think it's worth updating the entire thing for just one frame. I do hope to see HD versions of the Sonic Team logo and the DC intro though, I'll add support for them in DC Conversion if it happens.

**Comment by DisapprovingOwl** [comment]
Fair point on it being not worth updating just for that single frame - I'll backport the changes into my source for the reconstructed intro anyway, so that if for some reason any more substantial changes happen in future this tweak can be included at that point. I'll also port it over into my source for the (currently abandoned) full-frame 640x480 DC Intro - it's still not viable as far as I can see, but fixing the top/bottom of the Gamma shot using the AutoDemo assets plugs at least one of the gaps in the current version. You never know - it may become viable one day...! (Though that's largely dependent on finding a high-quality source for Tails' Japanese "character spotlight" advert, as that's the only place I've found a full-frame version of the version of Tails' shot seen in the DC Intro).

In the meantime - thanks so much for sharing the settings you used; much appreciated! Once I've finished the last remaining bit of work on the DX Gamma shot (and learned how to actually make a mod!) this'll allow me to create the fixed DX intro mod... I'm guessing most people will probably pass it up in favour of the HD version once that's completed and released, but I'd like to put it out there in case anyone wants it at the original resolution instead, haha. XD

On that note - great to hear you're planning on supporting the HD versions of the Sonic Team logo and DC intro; I'll be supplying the frames and information for them to one of the people working on the HD Cutscenes mod (along with my fixed version of the DX intro too), so if all goes to plan HD DC videos should be a part of it. = )
